### ISSUE TITLE
start.js: Unhandled Promise Rejection: SyntaxError: Invalid regular expression: invalid group specifier name

### ISSUE THREAD

**Issue by BillBuilt** [sveltejs/kit#4596]
### Describe the bug

Getting the error `Unhandled Promise Rejection: SyntaxError: Invalid regular expression: invalid group specifier name` in Safari only. The homepage loads, but anything beyond that is not working. I have  no idea when this started, I am just now to the point where I can start testing other browsers and certainly did not expect this. Works fine in Chrome (Brave) and Firefox. Not able to test windows just yet. I really don't know how to start debugging this so any help, pointers, advice is greatly appreciated! Thanks!

### Reproduction

Not likey able to offer any reporoduction steps since I really have no idea what this error means or implies. I have 0 errors or issues on other browsers, only Safari reports this.

### Logs

```shell
Unhandled Promise Rejection: SyntaxError: Invalid regular expression: invalid group specifier name

(note - i cannot copy the entire error trace from Safari, but it states this happens in line 1644 in start.js and this is the code surrounding that line - see attachments for full output)

if (hydrate) {
     await client._hydrate(hydrate); <<< this is line 1644
}
```

### System Info

```shell
System:
    OS: macOS 12.3.1
    CPU: (8) x64 Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
    Memory: 7.38 GB / 40.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 17.2.0 - ~/.nodenv/shims/node
    npm: 8.1.4 - ~/.nodenv/shims/npm
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.34 
    @sveltejs/kit: next => 1.0.0-next.308 
    svelte: ^3.44.0 => 3.46.6
```

### Severity

blocking all usage of SvelteKit

### Additional Information

Blocking at least for Safari Version 15.4 (17613.1.17.1.13)

Attached are the screenshots of the Safari console

<img width="1084" alt="Screen Shot 2022-04-12 at 9 54 00 AM" src="https://user-images.githubusercontent.com/28831382/162980779-43a35a8d-650f-4896-884a-11dfbccb4330.png">

<img width="776" alt="Screen Shot 2022-04-12 at 9 53 38 AM" src="https://user-images.githubusercontent.com/28831382/162980842-e827972a-02df-4f51-b591-3cda859870b1.png">

**Comment by rmunn** [comment]
According to a comment on https://stackoverflow.com/questions/51568821/works-in-chrome-but-breaks-in-safari-invalid-regular-expression-invalid-group (which I have not verified), Safari does not support lookbehind assertions in regexes. That might be a place to start looking: any recent changes to regexes.

**Comment by Conduitry** [comment]
As far as I can tell, SvelteKit itself doesn't use any lookbehinds, so if that's what's causing this, it would be in your application code or in some dependency.

Without a reproduction, that's about all anyone's going to be able to tell you.

**Comment by BillBuilt** [comment]
Yea I read about the lookbehind and Safari, I guess I was just confused as to why it was reporting the error as coming from `start.js` rather than the point of the actual regexp statement, if that were indeed the issue. I will continue with my efforts. Thanks for the pointers!

**Comment by BillBuilt** [comment]
I was able to track down the problematic regex. For anyone else that may happen on to this problem, I just searched all files for `?<=\` and then implemented a workaround so that it doesnt need to be used. Still wish it could have reported the error at the point of failure as oppsed to just `start.js`. I even did `npm run build` followed by `npm run preview` hoping that it would point me to the actual spot where the regex was located, but it only pointed me to the same place in the built code.
